modules/mongodb/mongodb.go

```diff
@@ -73,6 +73,31 @@ func WithPassword(password string) testcontainers.CustomizeRequestOption {
 	}
 }
 
+// WithReplicaSet configures MongoDB to start with a replica set named "rs".
+func WithReplicaSet() testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) {
+		req.Cmd = append(req.Cmd, "--replSet", "rs")
+		req.LifecycleHooks = append(req.LifecycleHooks, testcontainers.ContainerLifecycleHooks{
+			PostReadies: []testcontainers.ContainerHook{
+				func(ctx context.Context, c testcontainers.Container) error {
+					cIP, err := c.ContainerIP(ctx)
+					if err != nil {
+						return err
+					}
+
+					cmd := eval("rs.initiate({ _id: 'rs', members: [ { _id: 0, host: '%s:27017' } ] })", cIP)
```

Comment:
In my experience, the lack of the 'members' attribute occasionally makes the replica set [unreachable](https://www.mongodb.com/docs/manual/reference/replica-configuration/#mongodb-rsconf-rsconf.members-n-.host). Consequently, I consistently use this attribute whenever configuring replica sets. Given its usage in the provided documentation example, I consider it appropriate here as well.

Additionally, the documentation suggests using DNS only when the IP address can change. Since I don't fully understand the testcontainers source code, I'm unsure if this is a concern.

I can change them if necessary.